SConstruct

```diff
@@ -121,6 +121,10 @@ AddOption('--ignore-style', action='store_true',
           help='Disable style checking hooks')
 AddOption('--linker', action='store', default=None, choices=linker_options,
           help=f'Select which linker to use ({", ".join(linker_options)})')
+AddOption('--mold-path', action='store', default='/usr/bin/mold',
+          help='Path to the binary of mold, the linker. For GCC, the path must'
```

Comment:
Is binary, or is it some kind of directory? If it's just a binary then how does this line work:

```py
 (isdir('/usr/libexec/mold') and \
                           conf.CheckLinkFlag('-B/usr/libexec/mold')) or \
```

This is saying "/usr/libexec/mold" is directory.

Reply:
`/usr/bin/mold` should be a binary.
I'm not very familar with how mold is installed so I keep the original code there.

Reply:
> `/usr/bin/mold` should be a binary.
> I'm not very familar with how mold is installed so I keep the original code there.

Mold appears to be installed at `/usr/libexec/mold/ld` then symlinked to `/usr/bin/mold`. The `-B` flag is a way to tell the compiler where data files are... i don't know why this helps but apparently it was necessary for GCC <12. gcc-12 and beyond you only need to do the `-fuse-ld=mold` like everything else.

Reply:
Yeah, `-B $PREFIX` will let gcc finding `$PREFIX/ld`, so it should be a directory instead of a binary. I'll update this.